pta-intelligent-search-metadata-extractor/src/main/java/fi/maanmittauslaitos/pta/search/metadata/ISOMetadataExtractorConfigurationFactory.java

```diff
@@ -143,52 +145,227 @@ public DocumentProcessingConfiguration createMetadataDocumentProcessingConfigura
+						"/*/gmd:southBoundLatitude/gco:Decimal/text())"));
+
+		// East / right
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.EXTENT_EAST_BOUND,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(" +
+						XPATH_EXTENT_ROOT +
+						"/*/gmd:eastBoundLongitude/gco:Decimal/text())"));
+
+		// West / left
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.EXTENT_WEST_BOUND,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(" +
+						XPATH_EXTENT_ROOT +
+						"/*/gmd:westBoundLongitude/gco:Decimal/text())"));
+
+
+		// Classification
+		// List of possible values found at http://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas/resources/codelist/gmxCodelists.xml#MD_ClassificationCode
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.CLASSIFICATION,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(//gmd:identificationInfo/*/gmd:resourceConstraints/*/gmd:classification/*/@codeListValue)"));
+
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.DOWNLOAD_LINKS,
+				new DownloadLinksXmlCustomExtractor(),
+				"(//gmd:distributionInfo/*/gmd:transferOptions/*/gmd:onLine)"));
+
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.SERVICE_ASSOCIATED_RESOURCES,
+				new ServiceAssociatedResourcesXmlCustomExtractor(),
+				"(//gmd:identificationInfo/*/srv:operatesOn)"));
+
+
+		// Additional extractors for other documents
+		// A resources additional resources need to be queried in a separate query, so its included under a separate topic/group
+		// Associated resources
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.ADDITIONAL.ASSOCIATED_RESOURCES,
+				new AssociatedResourcesXmlCustomExtractor(),
+				"(//csw:SearchResults/csw:BriefRecord)"));
+
 		return configuration;
 	}
 
 
 	@Override
 	public DocumentProcessor createMetadataDocumentProcessor() throws ParserConfigurationException {
-		DocumentProcessingConfiguration configuration = createMetadataDocumentProcessingConfiguration();
-		return getDocumentProcessorFactory().createXmlProcessor(configuration);
+		return createMetadataDocumentProcessor(null);
 	}
 
-	private FieldExtractorConfiguration createXPathExtractor(String field, FieldExtractorType type, String xpath) {
-		FieldExtractorConfigurationImpl ret = new FieldExtractorConfigurationImpl();
-		ret.setField(field);
-		ret.setType(type);
-		ret.setQuery(xpath);
-
-		return ret;
-	}
-
-	private FieldExtractorConfiguration createXPathExtractor(String field, CustomExtractor extractor, String xpath) {
-		FieldExtractorConfigurationImpl ret = new FieldExtractorConfigurationImpl();
-		ret.setField(field);
-		ret.setType(FieldExtractorType.CUSTOM_CLASS);
-		ret.setQuery(xpath);
-		ret.setCustomExtractor(extractor);
-
-		return ret;
+	@Override
+	public DocumentProcessor createMetadataDocumentProcessor(DocumentProcessingConfiguration configuration) throws ParserConfigurationException {
+		if (configuration == null ) {
```

Comment:
See above. No null handling is necessary, just pass the configuratoin as is to the DocumentProcessorFactory